sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -1260,10 +1282,17 @@ internal async Task<BinaryData> GetStateInternal(TimeSpan timeout)
                 if (response.StatusCode == AmqpResponseStatusCode.OK)
                 {
                     var amqpMapList = response.GetListValue<AmqpMap>(ManagementConstants.Properties.Messages);
-                    foreach (var entry in amqpMapList)
+                    // not using foreach for better performance
+                    for (var index = 0; index < amqpMapList.Count; index++)
                     {
+                        var entry = amqpMapList[index];
+                        messages ??= amqpMapList is IReadOnlyCollection<AmqpMap> readOnlyList
```

Comment:
Should I add a comment here too why we are doing this for our future selfs?

Reply:
Yeah, that would be helpful.